### PR TITLE
feat: Add facets for several new resources

### PR DESCRIPTION
**Pull request by obs-gh-enricogiorio**
This facet was computed using OTTL's Len() on a slice of pointers, which is not a supported ValueType for Len().

Therefore, this facet will have to be computed with a custom processor.

### CODE REVIEW

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 1]
```diff
@@ -235,11 +254,13 @@ processors:
...
           - body["kind"] == "StatefulSet"
         statements:
+          - set(attributes["observe_transform"]["identifiers"]["statefulSetName"], body["metadata"]["name"])
+          - set(attributes["observe_transform"]["identifiers"]["statefulSetUid"], body["metadata"]["uid"])
           # status
           - set(attributes["observe_transform"]["facets"]["currentReplicas"], body["status"]["currentReplicas"])
           - set(attributes["observe_transform"]["facets"]["readyReplicas"], body["status"]["readyReplicas"])
           # spec
-          - set(attributes["observe_transform"]["facets"]["service"], body["spec"]["serviceName"])
+          - set(attributes["observe_transform"]["facets"]["serviceName"], body["spec"]["serviceName"])
```
Comment (xueweiz) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:263]: Good catch!

File: charts/agent/Chart.yaml [hunk 2]
```diff
@@ -2,30 +2,30 @@ apiVersion: v2
 name: agent
 description: Chart to install K8s collection stack based on Observe Agent
 type: application
-version: 0.8.0
+version: 0.8.1
 appVersion: "1.0.0"
 dependencies:
-  - name: opentelemetry-collector
```
Comment (xueweiz) [inline charts/agent/Chart.yaml:8]: the space edits seems weird, maybe your linter is different?

Comment (obs-gh-enricogiorio) [reply charts/agent/Chart.yaml:8]: Yeah I found a weird setting in my formatter that makes lists not indented. Weird.

**Review by arthur-observe — CHANGES_REQUESTED** [review]
There should be a Chart.lock update

if you run make lint from root then pre-commit run -a (twice) should get everything right

File: charts/agent/templates/_deployment-cluster-events-config.tpl [hunk 3]
```diff
@@ -226,24 +251,127 @@ processors:
...
+          - set(attributes["observe_transform"]["facets"]["type"], body["type"])
+      - context: log
+        conditions:
+          - body["kind"] == "PersistentVolume"
+        statements:
+          - set(attributes["observe_transform"]["identifiers"]["persistentVolumeName"], body["metadata"]["name"])
+          - set(attributes["observe_transform"]["identifiers"]["persistentVolumeUid"], body["metadata"]["uid"])
+          # spec
+          - set(attributes["observe_transform"]["facets"]["capacity"], body["spec"]["capacity"]["storage"])
+          - set(attributes["observe_transform"]["facets"]["persistentVolumeReclaimPolicy"], body["spec"]["persistentVolumeReclaimPolicy"])
```
Comment (colecorrente) [inline charts/agent/templates/_deployment-cluster-events-config.tpl:295]: for these facets, why prefix with "persistentVolume" 

ie why not just "volumeReclaimPolicy" and "volumeClass"


same for PersistentVolumeClaim below

**Pull request closed without merging**